spec/std/iterator_spec.cr

```diff
@@ -527,6 +541,10 @@ describe Iterator do
       iter.next.should eq([7, 8])
       iter.next.should be_a(Iterator::Stop)
     end
+
+    it "doesnt conflict with `::Slice` type" do
+      assert_iterates_iterator [1, 2, 3], MockIterator.new.each
+    end
```

Comment:
Does it makes sense, actually?

Reply:
Probably a bit overkill yea, but doesn't really hurt anything either. Maybe it'll prevent a regression in crystal 3.0 😅